content/blog/2019-01-31-rdhs.md

```diff
@@ -0,0 +1,787 @@
+
+
+```r
+# read in a key from a local file
+key <- sodium::hash(charToRaw(readLines("scripts/key.txt")))
+
+# create a tat with all the necessary login credentials
+zip("rdhs.json.tar",files=c("rdhs.json", "tests/testthat/rdhs.json"))
+
+# read this tar in as binary data 
+dat <- readBin("rdhs.json.tar",raw(),file.size("rdhs.json.tar"))
+
+# encrypt the data using sodium and our key before saving it
+enc <- sodium::data_encrypt(msg = dat,key = key)
+saveRDS(enc,"rdhs.json.tar.enc")
+```
+
+This could then be included in the GitHub repository, and I could then
+set up the key as a Travis environment variable to decrypt it. This decryption 
+step could then be written within my `.travis.yml` file, and would mean that all
+my tests had access to my login credentials in a secure way. 
+
+### Acknowledgements and Final Thoughts
+
+Firstly, I want to thank [Anna Krystalli](https://github.com/annakrystalli) for handling
+the review, and for being incredibly patient throughout, especially at the end as we
+were fixing the last authentication bug. Also many thanks to [Lucy McGowan](https://github.com/LucyMcGowan)
+and [Duncan Gillespie](https://github.com/dosgillespie) for taking the time to 
+review the package and for their input, which led to lots of improvements (and also
+linking the `add_line` function from `httr` was seriously helpful, and I've used 
+that function in lots of other my other work now). I also wanted to more broadly thank 
+the review process as a whole. Having the option to discuss the package and needed 
+solutions with the reviewers within a GitHub issues system is fantastic. It made the process 
+personal and was substantially improved over review processes I have had at academic journals.
+Lastly, another big thank you [Jeff Eaton](https://github.com/jeffeaton) and 
+[Rich FitzJohn](https://github.com/richfitz), and also to the infectious
+disease epidemiology department at Imperial for providing a lot of really helpful 
+guinea pig testing of the numerous iterations of `rdhs`.
+
+---
+
+### Options to Contribute
+
+There are a few things that would be great to add in the future.
+
+The first would be to add a suite of tools for doing spatial mapping. A lot of the
+time, people want to know what the prevalence of x is either at a fine spatial scale, 
+or grouped at administrative/county/state levels. `rdhs` helps provide the tools to
+get geolocated measures of x, and I think it would be a great next step to add
+a suite of mapping tools. It would be great if they could be used to either create a mesh 
+through these points (probably using `INLA`), or calculate survey weighted means at requested
+spatial scales or match them to a provided `SpatialPolygons` object. Related to this is
+it would be good to also link in the [Spatial Data Repository](https://spatialdata.dhsprogram.com/home/)
+from the DHS, so that users can easily download shape files for their analyses ([issue #71](https://github.com/ropensci/rdhs/issues/71).
```

Comment:
```suggestion
from the DHS, so that users can easily download shape files for their analyses ([issue #71](https://github.com/ropensci/rdhs/issues/71)).
```

added missing parenthesis